File: resume-analyzer-backend/utils/skillExtractor.js
const SKILLS = require("../ai/skills");

const extractSkills = (resumeText) => {
  const text = resumeText.toLowerCase();
  const extractedSkills = [];

  SKILLS.forEach(skill => {
    if (text.includes(skill)) {
      extractedSkills.push(skill);
    }
  });

  return extractedSkills;
};

module.exports = extractSkills;

File: ml-service/ats_engine/skill_extractor.py
import re
from ats_engine.skills_db import SKILLS

def extract_skills(text):
    text = text.lower()
    found_skills = set()

    for skill in SKILLS:
        pattern = r"\b" + re.escape(skill) + r"\b"
        if re.search(pattern, text):
            found_skills.add(skill)

    return list(found_skills)
